src/virtual-machine/parser/tokens/literals.ts

```diff
@@ -86,7 +95,38 @@ export class FunctionLiteralToken extends Token {
   }
 
   override compile(compiler: Compiler): Type {
-    //! TODO: Implement compiling the function body.
+    compiler.context.push_env()
+    const jump_instr = new JumpInstruction()
+    compiler.instructions.push(jump_instr)
+    const func_start = compiler.instructions.length
+    const block_instr = new FuncBlockInstruction(
+      this.signature.parameters.length,
+    )
+    compiler.instructions.push(block_instr)
+    compiler.type_environment = compiler.type_environment.extend()
+
+    let cnt = 0
+    for (const param of this.signature.parameters) {
+      const name = param.identifier || (cnt++).toString()
+      compiler.context.env.declare_var(name)
+      compiler.type_environment.addType(name, param.type.compile(compiler))
+    }
+
+    for (const sub_token of this.body.statements) {
+      sub_token.compile(compiler)
+      if (isExpressionToken(sub_token))
+        compiler.instructions.push(new PopInstruction())
```

Comment:
Just a note, we should change the parser to emit some kind of `ExpressionStatementToken` instead of `ExpressionToken` whenever an expression is used as a statement. Then the `ExpressionStatementToken`, when compiled, can clean up after itself.